File: c++ code/week1/examples/basic functions/basic functions/datatypeconversion.cpp
#include <opencv2/opencv.hpp>

int main()
{
	cv::Mat source;

	// scale will convert pixel values
	double scale = 1 / 255.0;
	double shift = 0;

	// read image
	source = cv::imread("../../../../../images/lena.jpg");

	// convert from usigned char to float-32 bit
	source.convertTo(source, CV_32FC3, scale, shift);

	// convert from float to unsigned char
	source.convertTo(source, CV_8UC3, 1.0 / scale, shift);

	return EXIT_SUCCESS;
}

File: python code/basic functions/datatypeconversion.py
import cv2
import numpy as np

# read image
source = cv2.imread("../../images/lena.jpg", flags=1)

scalingFactor = 1 / 255.0

# convert unsigned int to float
source = np.float32(source)
source = source * scalingFactor

# convert back to unsigned int
source = source * (1.0 / scalingFactor)
source = np.uint8(source)